src/views/SignInView.vue

```diff
@@ -0,0 +1,51 @@
+<script setup lang="ts">
+import { auth } from '@/firebase'
+import { FirebaseError } from 'firebase/app'
+import { signInWithEmailAndPassword } from 'firebase/auth'
+import { onMounted } from 'vue'
+import { useRoute, useRouter } from 'vue-router'
+import { getCurrentUser } from 'vuefire'
+
+const route = useRoute()
+const router = useRouter()
+
+// Get the next page from the 'redirect' query parameter
+const nextPage =
+  route.query.redirect && typeof route.query.redirect === 'string' ? route.query.redirect : '/'
+
+async function handleSignIn(payload: { email: string; password: string }) {
+  try {
+    await signInWithEmailAndPassword(auth, payload.email, payload.password)
+
+    // Send user to the next page
+    router.replace(nextPage)
+  } catch (error) {
+    if (error instanceof FirebaseError) {
+      // Handle Firebase Authentication error
+      console.error(error.code, error.message)
+    }
```

Comment:
When the error is a Firebase Error, something went wrong on sign-in. Maybe a bad password or missing users. You will want to handle these errors differently.